### ISSUE TITLE
How to modify Roberta's loss function

### ISSUE THREAD

**Issue by skye95git** [facebookresearch/fairseq#4020]
#### What is your question?
1. I pretrain Roberta with the Wikitext-103 dataset. I have used 8 A100s, and the task has not been finished for 2 days. How long does the pre-training task take?
2. If I pre-train Roberta using a code corpus, are the pre-processing steps the same as those in [readme](https://github.com/pytorch/fairseq/blob/main/examples/roberta/README.pretraining.md)?
3. I want to modify Roberta's loss function to add the `replace token detection` based on the `Masked language model`. But FairseQ is so wrapped that I can't find the location of the target function. I wonder if you could give me a hint?
4. If I modify the target function, do I need to modify the network structure accordingly?

**Comment by kscp123** [comment]
hello, have you solved the third problem?i also want to modify the loss function.

**Comment by kscp123** [comment]
hello, have you solved the third problem?i also want to modify the loss function.
